hazelcast/proxy/multi_map.py

```diff
@@ -54,37 +91,119 @@ def contains_entry(self, key, value):
+
+        **Warning:
+        This method uses hashCode and equals of the binary form of the key, not the actual implementations of hashCode
+        and equals defined in the key's class.**
+
+        **Warning-2:
+        The list is NOT backed by the multimap, so changes to the map are list reflected in the collection, and
+        vice-versa.**
+
+        :param key: (Object), the specified key.
+        :return: (List), the list of the values associated with the specified key.
+        """
         check_not_none(key, "key can't be None")
         key_data = self._to_data(key)
         return self._encode_invoke_on_key(multi_map_get_codec, key_data, key=key_data,
                                           thread_id=thread_id())
 
     def is_locked(self, key):
+        """
+        Checks the lock for the specified key. If the lock is acquired then returns true, else returns false.
+
+        **Warning:
+        This method uses GetHashCode and Equals of binary form of the key, not the actual implementations of GetHashCode
+        and Equals defined in key's class.**
+
+        :param key: (Object), the key that is checked for lock
+        :return: (Bool), true if lock is acquired, false otherwise.
+        """
         check_not_none(key, "key can't be None")
         key_data = self._to_data(key)
         return self._encode_invoke_on_key(multi_map_is_locked_codec, key_data, key=key_data)
 
     def force_unlock(self, key):
+        """
+        Releases the lock for the specified key regardless of the lock owner. It always successfully unlocks the key,
+        never blocks, and returns immediately.
+
+        **Warning:
+        This method uses GetHashCode and Equals of binary form of the key, not the actual implementations of GetHashCode
+        and Equals defined in key's class.**
+
+        :param key: (Object), the key to lock.
+        """
         check_not_none(key, "key can't be None")
         key_data = self._to_data(key)
         return self._encode_invoke_on_key(multi_map_force_unlock_codec, key_data, key=key_data)
 
     def key_set(self):
+        """
+        Returns the list of keys in the multimap.
+
+        **Warning:
+        The list is NOT backed by the map, so changes to the map are NOT reflected in the list, and vice-versa.**
+
+        :return: (List), a list of the clone of the keys.
+        """
         return self._encode_invoke(multi_map_key_set_codec)
 
     def lock(self, key, lease_time=-1):
+        """
+        Acquires the lock for the specified key infinitely or for the specified lease time if provided.
+
+        If the lock is not available, then the current thread becomes disabled for thread scheduling purposes and lies
+        dormant until the lock has been acquired.
+
+        Scope of the lock is this map only. Acquired lock is only for the key in this map
+
+        Locks are re-entrant so if the key is locked N times then it should be unlocked N times before another thread
```

Comment:
Locks are re-entrant; so, if the key is locked N times, it should be unlocked N times before another thread..